frontends/ol-components/src/components/RoutedDrawer/RoutedDrawer.tsx

```diff
@@ -62,15 +63,20 @@ const RoutedDrawer = <K extends string, R extends K = K>(
   }, [requiredArePresent, setOpen, requiredParams])
 
   const removeUrlParams = useCallback(() => {
-    setSearchParams((current) => {
+    const getNewParams = (current: string) => {
       const newSearchParams = new URLSearchParams(current)
       params.forEach((param) => {
         newSearchParams.delete(param)
       })
       return newSearchParams
+    }
+    const newParams = getNewParams(location.search)
+    setSearchParams(newParams)
```

Comment:
`navigate` handles the navigation now, so you can get rid of `setSearchParams`. I would get rid of `useSearcParams` all together:

```diff
diff --git a/frontends/ol-components/src/components/RoutedDrawer/RoutedDrawer.tsx b/frontends/ol-components/src/components/RoutedDrawer/RoutedDrawer.tsx
index 7dcce5b47..a6eea26bf 100644
--- a/frontends/ol-components/src/components/RoutedDrawer/RoutedDrawer.tsx
+++ b/frontends/ol-components/src/components/RoutedDrawer/RoutedDrawer.tsx
@@ -71,12 +71,11 @@ const RoutedDrawer = <K extends string, R extends K = K>(
       return newSearchParams
     }
     const newParams = getNewParams(location.search)
-    setSearchParams(newParams)
     navigate({
       ...location,
       search: newParams.toString(),
     })
-  }, [setSearchParams, params, navigate, location])
+  }, [params, navigate, location])

   return (
     <Drawer
```